shared/truck-models/truck.go

```diff
@@ -0,0 +1,144 @@
+	// ErrMissingBrand error when truck brand is missing
+	ErrMissingBrand = errors.New("missing brand parameter")
+	// ErrMissingModel error when truck_model is missing
+	ErrMissingModel = errors.New("missing model parameter")
+	// ErrMissingMileague error when truck mileague is missing
+	ErrMissingMileague = errors.New("missing mileague parameter")
+	// ErrMissingLocation error when truck location is missing
+	ErrMissingLocation = errors.New("missing location parameter")
+	// ErrMissingAvailable error when truck avaliability is missing
+	ErrMissingAvailable = errors.New("missing available parameter")
+	// ErrInvalidAvailable error when an invalid truck available param is given
+	ErrInvalidAvailable = errors.New("invalid available parameter")
+	// ErrMissingTruckType error when an invalid truck type is given
+	ErrMissingTruckType = errors.New("missing truck type parameter")
+	// ErrInvalidYear error when an invalid truck year is given
+	ErrInvalidYear = errors.New("invalid year parameter")
+	// ErrInvalidMileague error when an invalid truck mileague is given
+	ErrInvalidMileague = errors.New("invalid mileague parameter")
+	// ErrInvalidTruckType error when an invalid truck type is given
+	ErrInvalidTruckType = errors.New("invalid truck type parameter")
+	// ErrInvalidLocation error when an invalid truck location is given
+	ErrInvalidLocation = errors.New("invalid truck location")
+)
+
+// TruckType is the type handler for the different types of trucks
+type TruckType string
+
+const (
+	// TruckTypeRigid is a truck of type rigid
+	TruckTypeRigid TruckType = "rigid"
+	// TruckTypeArticulated is a truck of type Articulated
+	TruckTypeArticulated TruckType = "articulated"
+	// TruckTypeTrailer is a truck of type Trailer
+	TruckTypeTrailer TruckType = "trailer"
+	// TruckTypeHighway is a truck of type Highway
+	TruckTypeHighway TruckType = "highway"
+)
+
+var (
+	// ValidTruckTypes is the map of all the valid truck types
+	ValidTruckTypes = map[TruckType]bool{
+		TruckTypeRigid:       true,
+		TruckTypeArticulated: true,
+		TruckTypeTrailer:     true,
+		TruckTypeHighway:     true,
+	}
+)
+
+// Truck is the struct handler for a truck
+type Truck struct {
+	TruckID           string       `json:"truck_id"`
+	PartnerID         string       `json:"partner_id"`
+	RegistrationPlate string       `json:"registration_id"`
+	Brand             string       `json:"brand"`
+	Model             string       `json:"model"`
+	Year              int          `json:"year"`
+	Mileague          int          `json:"mileague"`
+	CompletedTrips    int          `json:"completed_trips"`
+	Available         bool         `json:"available"`
+	Type              TruckType    `json:"type"`
+	Location          trips.Region `json:"location"`
+	CreationDate      time.Time    `json:"creation_date"`
+	UpdateDate        time.Time    `json:"update_date"`
+}
+
+// NewTruck returns a cleaned Truck structure with given values
+func NewTruck(truck Truck) (*Truck, error) {
```

Comment:
There is a smarter way of doing this  jeje, but I will do it myself

Reply:
Lol, I asked you if there was a better way of doing it. But awesome, I'll see how you do it